src/bbb/models/bnn.py

```diff
@@ -374,6 +374,27 @@ def prune_weights(self, weight_fill_masks: List[Tensor]):
                 layer.w_var_post.rho.masked_fill(weight_fill_masks[i], rho_replacement)
             )
 
+    def get_pruned_weight_samples(self, weight_fill_masks):
+        """ Returns the samples of weights that are active post-pruning """
+        bfc_layers = [layer for layer in self.model if isinstance(layer, BaseBFC)]
+        non_pruned_samples = []
+    
+        weight_samples = self.weight_samples()
+        for layer_samples, layer, layer_mask in zip(weight_samples, bfc_layers, weight_fill_masks):
+        
+            ls = torch.reshape(layer_samples, shape=(self.inference_samples, layer.dim_in, layer.dim_out))
+            non_pruned_samples_layer = []
+        
+            for i, sample in enumerate(ls):
+                non_pruned_weights = sample[~layer_mask.T]
```

Comment:
How come we need to transpose the mask? Seems strange...

Reply:
The order of the mask is wrong, the dimensions were flipped (mask was 1200 x 786) whereas dim_in x dim_out = (786x1200)

Reply:
Would be good to understand _how_ the order of the mask ended up being wrong; I don't reshape at any point as far as I remember.

Reply:
Yeah, the weights are (dim_out, dim_in)

![image](https://user-images.githubusercontent.com/32013626/160256393-f1e942a9-16a7-4404-a7c5-06c94b4bcf3f.png)